weaviate/util.py

```diff
@@ -772,3 +777,34 @@ def _type_request_response(json_response: Any) -> Optional[Dict[str, Any]]:
         return None
     assert isinstance(json_response, dict)
     return json_response
+
+
+def _decode_json_response_dict(
+    response: requests.Response, location: str
+) -> Optional[Dict[str, Any]]:
+    if response is None:
+        return None
+
+    if response.status_code == 200:
+        try:
+            json_response = response.json()
+            return json_response
+        except JSONDecodeError:
+            raise ResponseCannotBeDecodedException(location, response)
+
+    raise UnexpectedStatusCodeException(location, response)
+
+
+def _decode_json_response_list(
```

Comment:
This is exactly the same as ` _decode_json_response_dict`, why not to have a single function?

Reply:
different return type, otherwise we'd need to type check each response if it is dict or list

Reply:
BTW we did not check this before, so why we would have to check it now?

Reply:
got it, thanks!